### PR TITLE
Fix OpenSSL public ECDSA keys import compatibility

### PR DESCRIPTION
**Pull request by gsstoykov**
**Description**:
This PR introduces basic logics for working with ASN1 formed data. SDK should support imports from DER/PEM format
and should be fully OpenSSL compatible. SDK now handles importing different types of EC Public Keys. 

**Related issue(s)**:

Fixes  https://github.com/hashgraph/hedera-sdk-cpp/issues/619 https://github.com/hashgraph/hedera-sdk-cpp/issues/423

**Notes for reviewer**:

**Checklist**

- [X] Documented (Code comments, README, etc.)
- [X] Tested (unit, integration, etc.)

### CODE REVIEW

File: sdk/main/include/impl/ASN1ECPublicKey.h [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+   * @brief Get the key value associated with the ASN.1 elliptic curve public key.
+   * @return The key as a vector of bytes.
+   */
+  const std::vector<std::byte> getKey() const override;
+
+private:
+  /**
+   * @brief Constructor for ASN.1 elliptic curve public key.
+   */
+  ASN1ECPublicKey(){};
```
Comment (rwalworth) [inline sdk/main/include/impl/ASN1ECPublicKey.h:83]:
```suggestion
  ASN1ECPublicKey() = default;
```

File: sdk/main/include/impl/ASN1ECPublicKey.h [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+   * @brief Constructor for ASN.1 elliptic curve public key from a vector of bytes.
+   * @param bytes The vector of bytes containing the ASN.1 key data.
+   */
+  ASN1ECPublicKey(const std::vector<std::byte>& bytes);
+
+  /**
+   * @brief Get the key value associated with the ASN.1 elliptic curve public key.
+   * @return The key as a vector of bytes.
+   */
+  const std::vector<std::byte> getKey() const override;
```
Comment (rwalworth) [inline sdk/main/include/impl/ASN1ECPublicKey.h:77]:
```suggestion
  std::vector<std::byte> getKey() const override;
```

File: .gitignore [hunk 3]
```diff
@@ -888,3 +888,6 @@ package
 
 ### Ignore environment file
 .env
+
+### VCPKG install
+vcpkg_installed/
```
Comment (rwalworth) [inline .gitignore:893]: The `vcpkg_installed` should be going in the `build` folder, are you seeing otherwise? This shouldn't be needed

**Review by rwalworth — COMMENTED** [review]
This needs to be ran through `clang-format`, there are a few areas that don't have proper formatting

File: sdk/main/src/ECDSAsecp256k1PublicKey.cc [hunk 4]
```diff
@@ -419,8 +400,7 @@ std::vector<std::byte> ECDSAsecp256k1PublicKey::toBytesRaw() const
   }
 
   // Don't return the algorithm identification bytes, and compress
-  return compressBytes({ publicKeyBytes.cbegin() + static_cast<long>(DER_ENCODED_UNCOMPRESSED_PREFIX_BYTES.size()),
-                         publicKeyBytes.cend() });
+  return { std::vector<std::byte>(publicKeyBytes.cbegin() + internal::asn1::ASN1_CPUBK_PREFIX_BYTES.size() + 1, publicKeyBytes.cend())};
```
[the remark this answers is not in the document]
Comment (gsstoykov) [reply sdk/main/src/ECDSAsecp256k1PublicKey.cc:403]: The publicKeyBytes are returned from OpenSSL. OpenSSL EVP_KEY would always hold a compressed Public Key in this implementation. It is the most simple form to hold the Public Key. I decided this as the Y coord of the Public Key is derivable by the X coord.

**Review by rwalworth — APPROVED** [review]
LGTM

**Pull request merged**